packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -194,6 +194,18 @@ ruleTester.run('prefer-optional-chain', rule, {
     'match && match$1 !== undefined;',
     'foo !== null && foo !== undefined;',
     "x['y'] !== undefined && x['y'] !== null;",
+    // private properties
+    'this.#a && this.#b;',
```

Comment:
This should still report a complaint:

```ts
class Violation {
  #property = 0;

  test() {
    const self = Math.random() ? this : undefined;

    self && self#.property;
  }
}
```

I suspect the issue is returning `''` instead of the actual node text above.

Reply:
bump on this 😇

Reply:
The highlighted line has two different properties and the example seems to be covered by this, am I missing something?
https://github.com/typescript-eslint/typescript-eslint/blob/0afc516667c81d5b6f9c8ea09f7d6415bccc987d/packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts#L141-L144

Reply:
Ah sorry you're right, I missed that they're the same property. But this actually brings up that there's a bug in this PR for private identifiers: https://github.com/microsoft/TypeScript/issues/42734 blocks them from being used.

https://deploy-preview-6028--typescript-eslint.netlify.app/play#ts=4.9.3&sourceType=module&code=MYGwhgzhAEBqCWB7cAXJA7aBvAUNaAxAA4BOiRApiSgJ7QC80ADANw57QoUQoAUAlNg75gidD2gQKIAGYNoAWTAoAFgDoSYdABNEAWwHQA-JxXwYALmgBXHRRnx0FbW2GTpMo2uJlK1Gmz4AL44IUA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

```ts
class Violation {
  #property = 0;

  test() {
    const self = Math.random() ? this : undefined;

    self && self.#property;
  }
}
```

Reply:
So do I revert it for now? What do you suggest?

Reply:
Well, it shouldn't throw an error - that's definitely a bug. But it also shouldn't complain. The rule should effectively ignore this case for now.

Reply:
Kept the private properties in computed nodes